composer/trainer/trainer.py

```diff
@@ -403,12 +426,26 @@ def create_from_hparams(cls, hparams: TrainerHparams) -> Trainer:
         train_dataloader = hparams.train_dataset.initialize_object(train_device_batch_size, hparams.dataloader)
 
         eval_device_batch_size = hparams.eval_batch_size // dist.get_world_size()
-        if hparams.val_dataset.shuffle and hparams.eval_subset_num_batches is not None:
-            warnings.warn(
-                textwrap.dedent(f"""SubsetNumBatchesWarning: When specifying eval_subset_num_batches,
-            (set to {hparams.eval_subset_num_batches}), val_dataset.shuffle should be set to False. Otherwise,
-            each evaluation epoch may load a different subset of samples."""))
-        eval_dataloader = hparams.val_dataset.initialize_object(eval_device_batch_size, hparams.dataloader)
+        if hparams.val_dataset is not None:
+            if hparams.val_dataset.shuffle and hparams.eval_subset_num_batches:
+                warnings.warn(
+                    textwrap.dedent(f"""SubsetNumBatchesWarning: When specifying eval_subset_num_batches,
+                (set to {hparams.eval_subset_num_batches}), val_dataset.shuffle should be set to False. Otherwise,
+                each evaluation epoch may load a different subset of samples."""))
+            eval_dataloader = hparams.val_dataset.initialize_object(eval_device_batch_size, hparams.dataloader)
+
+        if hparams.evaluators is not None and len(hparams.evaluators) > 0:
+            eval_dataloader = [
+                evaluator.initialize_object(eval_device_batch_size, hparams.dataloader)
+                for evaluator in hparams.evaluators
+            ]
+            for evaluator in hparams.evaluators:
+                if evaluator.eval_dataset.shuffle and hparams.eval_subset_num_batches:
+                    warnings.warn(
+                        textwrap.dedent(f"""SubsetNumBatchesWarning: When specifying eval_subset_num_batches,
+                    (set to {hparams.eval_subset_num_batches}), evaluator.dataloader.shuffle should be set to False. Otherwise,
```

Comment:
Can you add context here to which evaluator this warning is for? And then remove the break?